src/test/java/net/snowflake/client/providers/TimezoneProvider.java

```diff
@@ -0,0 +1,33 @@
+package net.snowflake.client.providers;
+
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+
+public class TimezoneProvider extends SnowflakeArgumentsProvider {
+  private int length;
+
+  private static List<Arguments> timeZones =
+      Arrays.asList(
+          Arguments.of("UTC"),
+          Arguments.of("America/Los_Angeles"),
+          Arguments.of("America/New_York"),
+          Arguments.of("Pacific/Honolulu"),
+          Arguments.of("Asia/Singapore"),
+          Arguments.of("CET"),
+          Arguments.of("GMT+0200"));
+
+  public TimezoneProvider(int length) {
+    this.length = length;
+  }
+
+  public TimezoneProvider() {
+    this.length = timeZones.size();
+  }
+
+  @Override
+  protected List<Arguments> rawArguments(ExtensionContext context) {
+    return timeZones.subList(0, length);
```

Comment:
we need length to just copy the list? cannot we just keep unmodifiable list in this class?

Reply:
The idea here was to allow for some tests to easily get just a few of this. There were some long running tests that were originally running with 3-4 timezones only, and I decided not to double their execution time. Having said that, this is not useful when using it directly in the `@ArgumentsSource` annotation, only when using the `cartesianProduct` util